bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java

```diff
@@ -0,0 +1,297 @@
+    private TellstickLocalSensorsDTO sensorList = null;
+    private TelldusLocalDeviceController controller = null;
+    private List<DeviceStatusListener> deviceStatusListeners = Collections.synchronizedList(new ArrayList<>());
+    private final HttpClient httpClient;
+    private ScheduledFuture<?> pollingJob;
+    /**
+     * Use cache for refresh command to not update again when call is made within 10 seconds of previous call.
+     */
+    private final ExpiringCache<Boolean> refreshCache = new ExpiringCache<>(Duration.ofSeconds(10),
+            this::refreshDeviceList);
+
+    public TelldusLocalBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        TelldusLocalConfiguration configuration = getConfigAs(TelldusLocalConfiguration.class);
+        this.controller = new TelldusLocalDeviceController(configuration, httpClient);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::refreshDeviceList, 11, configuration.refreshInterval,
+                TimeUnit.MILLISECONDS);
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (this.controller != null) {
+            this.controller.dispose();
+        }
+        deviceList = null;
+        sensorList = null;
+        super.dispose();
+    }
+
+    private boolean refreshDeviceList() {
+        try {
+            updateDevices(deviceList);
+            updateSensors(sensorList);
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } catch (TellstickException | InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        return false;
+    }
+
+    private synchronized void updateDevices(TellstickLocalDevicesDTO previouslist)
+            throws TellstickException, InterruptedException {
+        TellstickLocalDevicesDTO newList = controller
+                .callRestMethod(TelldusLocalDeviceController.HTTP_LOCAL_API_DEVICES, TellstickLocalDevicesDTO.class);
+        logger.trace("Device list {}", newList.getDevices());
+        if (newList.getDevices() != null) {
+            if (previouslist == null) {
+                logger.trace("updateDevices, Creating devices.");
+                for (TellstickLocalDeviceDTO device : newList.getDevices()) {
+                    device.setUpdated(true);
+                    synchronized (deviceStatusListeners) {
+                        for (DeviceStatusListener listener : deviceStatusListeners) {
+                            listener.onDeviceAdded(getThing(), device);
+                        }
+                    }
+                }
+                this.deviceList = newList;
+            } else {
+                logger.trace("updateDevices, Updating devices.");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Fixed!